### ISSUE TITLE
systemd-sleep hooks no longer receive action name

### ISSUE THREAD

**Issue by grawity** [systemd/systemd#12782]
**systemd version the issue has been seen with**
 > [242.29](https://git.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/systemd&id=92fa157c40324aa7c9d9526366dc66403b0237c8)

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > /usr/lib/systemd/system-sleep hooks should receive two arguments: 1) "pre" or "post", 2) the action such as "suspend" or "hibernate"

**Unexpected behaviour you saw**
 > The 2nd argument is an empty string.

**Steps to reproduce the problem**

1. Have any kind of executable hook script at /usr/lib/systemd/systemd-sleep/foo
2. Suspend the system

For example, this is what systemd executes when I trigger a normal suspend to RAM (systemctl suspend):

```
$ sudo strace -p 1 -f -e execve -s 1000
strace: Process 1 attached
strace: Process 15671 attached
[pid 15671] execve("/usr/lib/systemd/systemd-sleep", ["/usr/lib/systemd/systemd-sleep", "suspend"], 0x55aae8819060 /* 9 vars */) = 0
strace: Process 15672 attached
strace: Process 15673 attached
[pid 15673] execve("/usr/lib/systemd/system-sleep/rtcwake.sh", ["/usr/lib/systemd/system-sleep/rtcwake.sh", "pre", ""], 0x7ffee69b7110 /* 9 vars */) = 0
```

However, the journal entries correctly contain `SLEEP=suspend`, and it comes from exactly the same *arg_path* variable. Does it disappear somewhere inside execute_directories()?

**Comment by poettering** [comment]
Fix waiting in #12809

**Comment by NICHOLAS85** [comment]
I seem to be experiencing this problem on my Debian Testing system with systemd 242-5 installed. The second string is empty and scripts in `/lib/systemd/systemd-sleep/` are not executing correctly

```
$ sudo strace -p 1 -f -e execve -s 1000
-snip-
[pid 15132] execve("/lib/systemd/systemd-sleep", ["/lib/systemd/systemd-sleep", "suspend"], 0x56514fba4220 /* 13 vars */) = 0
strace: Process 15133 attached
strace: Process 15134 attached
[pid 15134] execve("/usr/lib/systemd/system-sleep/hdparm", ["/usr/lib/systemd/system-sleep/hdparm", "pre", ""], 0x7fffdbc30c30 /* 13 vars */strace: Process 15135 attached
) = 0
strace: Process 15136 attached
[pid 15135] execve("/usr/lib/systemd/system-sleep/refindhibernate.sh", ["/usr/lib/systemd/system-sleep/refindhibernate.sh", "pre", ""], 0x7fffdbc30c30 /* 13 vars */ <unfinished ...>
[pid 15136] execve("/usr/lib/systemd/system-sleep/unattended-upgrades", ["/usr/lib/systemd/system-sleep/unattended-upgrades", "pre", ""], 0x7fffdbc30c30 /* 13 vars */ <unfinished ...>
-snip-
```

This began occuring around 1-2 weeks ago but I am unsure which version introduced the problem. It occured in v242-4 aswell to my knowledge.
